R/create_covariates.R

```diff
@@ -0,0 +1,1264 @@
+#' Covariate SAGA raster covarates from base digital terrain dataset.
+#'
+#'#' Takes a dem and via SAGA GIS generates raster covariates at the resolution of
+#' the dem.
+#'
+#' Depending on your system the path to `saga_cmd` may need to be specified.
+#'
+#' @param dtm a character or file path location to raster with elevation data.
+#'        Outputs will be produced at the same resolution and extent as input dtm
+#' @param saga_path a character of the file to the SAGA directory on the analysts system.
+#'          On linux systems with SAGA GIS installed Use `SAGApath = ""`
+#' @param out_dir file.path. Directory where covariates will be written. If not
+#'     specified uses the default from the `fid` folder structure. If files already
+#'     exists they will NOT be overwritten.
+#' @param layers character vector. Covariates to be created. Default is \code(`all`)
+#'  - call \code(`PEMprepr::layer_options`) for full list.
+#'
+#' @return TRUE
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' #  # start testing
+#' #dtm = fs::path(PEMprepr::read_fid()$dir_1020_covariates$path_rel,"25m","dem.tif")
+#' #  saga_path = "C:\\Programs\\saga-9.2.0_x64\\saga-9.2.0_x64"
+#' #  out_dir = PEMprepr::read_fid()$dir_1020_covariates$path_rel
+#' #  layers = "all"
+#' #  # end testing
+#' #--- load in dem and write to tempfile ---#
+#' aoi_raw <- system.file("extdata", "aoi.gpkg", package ="PEMprepr")
+#' aoi_raw <- sf::st_read(aoi_raw)
+#' aoi <- PEMprepr::aoi_snap(aoi_raw, "shrink")
+#' t25 <- create_template(aoi, 25)
+#' trim_raw <- cded_raster(aoi)
+#' trim <- terra::rast(trim_raw)
+#' dtm <- terra::project(trim, t25)
+#' tmp <- tempfile(fileext = ".tif")
+#' writeRaster(dtm, tmp)
+#'
+#'
+#' #--- create all SAGA covariates ---#
+#' create_covariates(dtm = tmp, layers = "all", output = dir, SAGApath = SAGApath)
+#'}
+
+create_covariates <- function(dtm = NULL,
+                              saga_path = NULL,
+                              out_dir = PEMprepr::read_fid()$dir_1020_covariates$path_rel,
+                              layers = "all"){
+
+  #  # start testing
+  # dtm = fs::path(PEMprepr::read_fid()$dir_1020_covariates$path_rel,"25m","dem.tif")
+  #  saga_path = "C:\\Programs\\saga-9.2.0_x64\\saga-9.2.0_x64"
+  #  out_dir = PEMprepr::read_fid()$dir_1020_covariates$path_rel
+  #  layers = "all"
+  #  # end testing
+
+
+  # internal data for package
+    data("layer_options", envir=environment())
+    data("moddir", envir=environment())
+    data("artifacts", envir=environment())
```

Comment:
Where are these datasets coming from? I see `R/sysdata.rda`, but don't see how it's created. Can we have a `data-raw` folder containing scripts/source data to reproducibly create it?

Reply:
This code was developed by someone else, so I'm not 100% sure I want to keep this format. They function as a filter and check for anomalies when covariates are created by tiles (ie some covariates are dependent on larger scales than a tile and artifacts are created when tiling). 

Perhaps we can discuss this one in more detail. I don't like have undocumented data creation, however they were created from a csv conversion to .R (I suspect).